server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -132,6 +146,30 @@ public ClusterState execute(ClusterState currentState) {
                 return allocationService.reroute(updatedState, "restored from remote store");
             }
 
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap = new HashMap<>();
+                boolean metadataFromRemoteStore = (request.clusterUUID() == null
+                    || request.clusterUUID().isEmpty()
+                    || request.clusterUUID().isBlank()) == false;
+                if (metadataFromRemoteStore) {
+                    // TODO integrate with download flow
+                    // something like RemoteClusterStateService.getLatestIndexMetadata()
+                    // full integration PR used for testing - https://github.com/soosinha/OpenSearch/pull/2/files
+                } else {
+                    for (String indexName : request.indices()) {
```

Comment:
Index patterns are not supported yet in this API?

Reply:
not yet, but we are working on it. For full cluster restore, we would be ignore the passed indices as well, as we are restoring all indices present in remote cluster state